neuroanalysis/stimuli.py

```diff
@@ -270,6 +270,27 @@ def get_stimulus_class(cls, name):
             raise KeyError('Unknown stimulus class "%s"' % name)
         return cls._subclasses[name]
 
+class LazyLoadStimulus(Stimulus):
```

Comment:
We could do that, but I couldn't think of a way to have lazy loading at the same time. I was copying MIESStimulus and I didn't know whether the lazy loading aspect of it was important or not.

Reply:
You implemented lazy loading in the other data model classes by deferring to a loader; let's just do the same for Stimulus (we shouldn't need LazyLoadStimulus any more than we need LazyLoadTSeries)

Reply:
If I remember right, the problem I ran into that led me to LazyLoadStimulus is that Stimulus isn't derived from the Container class like the rest of the model classes are and thus doesn't have a top_parent function or a way to get back to the loader. I implemented lazy loading because someone went to some trouble to do so in the subclassed data models in neuroanalysis/miesnwb.py, so I thought it might be important. But if we don't need lazy loading, it'd definitely simplify things to get rid of it.

Reply:
When I profile LazyLoadStimulus vs Stimulus I get quite a difference in speed. For a MP dataset with 92 sync_recordings, 4 recordings each (368 in total) it takes about 350 ms to access all the stimulus descriptions using LazyLoadStimulus, but takes about 6.5 sec to access all the stimulus descriptions using Stimulus (where we have to parse and load stimulus items when creating the stimulus). Neither of these numbers includes the time it takes to touch and make all the Recordings (which takes just under 2 seconds). Given that in the DatasetPipelineModule (both for multipatch and opto) we access recording.stimulus.description, but not the stimulus items, I would argue for keeping LazyLoadStimulus.